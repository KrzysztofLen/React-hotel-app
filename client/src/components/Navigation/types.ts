export type Props = {
  readonly to: string;
  readonly activeOnlyWhenExact?: boolean;
};

export type State = {
  isUnderBreakpoint: boolean;
};
