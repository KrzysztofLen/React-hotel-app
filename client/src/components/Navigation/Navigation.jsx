//@Flow
import React, {Component} from 'react';
import {NavLink, Route} from 'react-router-dom';
import {Logo} from "../Logo/Logo";
import Messages from "../External/Messages/Messages";

interface IProps {
	navlinks: Array<Object>
}

interface IState {
	isUnderBreakpoint: boolean
}

const SideMenuLink = ({ to, activeOnlyWhenExact }) => {
	return (
		<Route path={to}
		       exact={activeOnlyWhenExact}
		       children={({ match }, index) => (
			       <li key={index} className={match ? "side-navigation__item side-navigation__item--active" : "side-navigation__item"}>
				       <NavLink to={to} className={"side-navigation__link"}/>
			       </li>
		       )}
		/>
	);
};

class Navigation extends Component<IState> {
	constructor(props: IProps) {
		super(props);

		this.state = {
			isUnderBreakpoint: false
		};
	}

	onCheckResolution = () => {
		this.setState({ isUnderBreakpoint: window.innerWidth < 480 });
	}

	componentDidMount() {
		window.addEventListener("resize", () => {
			this.onCheckResolution();
		});

		this.onCheckResolution();
	}

	render() {
		return (
			<React.Fragment>
				{this.state.isUnderBreakpoint &&
				<Messages type={"danger"}
				          message={"Your device resolution is under supported value. Some content may not appear correctly"}/>
				}
				<nav className={"side-navigation"}>
					<Logo/>
					<ul className={"side-navigation__container"}>
						<SideMenuLink activeOnlyWhenExact={true} to="/" />
						<SideMenuLink to="/buy" />
						<SideMenuLink to="/reservation" />
						<SideMenuLink to="/add" />
					</ul>
				</nav>
			</React.Fragment>
		)
	}
}

export default Navigation;