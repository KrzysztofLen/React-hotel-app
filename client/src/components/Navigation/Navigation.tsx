import * as React from 'react';
import { NavLink, Route } from 'react-router-dom';
import { Logo } from '../Logo/Logo';
import Messages from '../External/Messages/Messages';

interface IProps {
  to: string;
  activeOnlyWhenExact?: boolean;
}

export const SideMenuLink = ({ to, activeOnlyWhenExact }: IProps) => {
  return (
    <Route
      path={to}
      exact={activeOnlyWhenExact}
      children={({ match }: any, index: number) => (
        <li
          key={index}
          className={
            match
              ? 'side-navigation__item side-navigation__item--active'
              : 'side-navigation__item'
          }>
          <NavLink to={to} className={'side-navigation__link'} />
        </li>
      )}
    />
  );
};

interface IState {
  isUnderBreakpoint: boolean;
}

class Navigation extends React.Component<{}, IState> {
  state = {
    isUnderBreakpoint: false,
  };

  onCheckResolution = () => {
    this.setState({ isUnderBreakpoint: window.innerWidth < 480 });
  };

  componentDidMount() {
    window.addEventListener('resize', () => {
      this.onCheckResolution();
    });

    this.onCheckResolution();
  }

  render(): JSX.Element {
    return (
      <React.Fragment>
        {this.state.isUnderBreakpoint && (
          <Messages
            type={'danger'}
            message={
              'Your device resolution is under supported value. Some content may not appear correctly'
            }
          />
        )}
        <nav className={'side-navigation'}>
          <Logo />
          <ul className={'side-navigation__container'}>
            <SideMenuLink activeOnlyWhenExact={true} to="/" />
            <SideMenuLink to="/buy" />
            <SideMenuLink to="/reservation" />
            <SideMenuLink to="/add" />
          </ul>
        </nav>
      </React.Fragment>
    );
  }
}

export default Navigation;
