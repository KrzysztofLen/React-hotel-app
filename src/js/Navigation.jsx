import React, {Component} from 'react';

class HotelLink extends Component {
	render() {
		return (
			<li className="side-nav__item side-nav__item--active">
				<a href="#" className="side-nav__link">
					<svg className="side-nav__icon">
						<use xlinkHref="img/sprite.svg#icon-home"></use>
					</svg>
					<span>Hotel</span>
				</a>
			</li>
		)
	}
}

const Navigation = () => {
	return (
		<nav className="sidebar">
			<ul className="side-nav">
				<li className="side-nav__item side-nav__item--active">
					<a href="#" className="side-nav__link">
						<svg className="side-nav__icon">
							<use xlinkHref="img/sprite.svg#icon-home"></use>
						</svg>
						<span>Hotel</span>
					</a>
				</li>
				<li className="side-nav__item">
					<a href="#" className="side-nav__link">
						<svg className="side-nav__icon">
							<use xlinkHref="img/sprite.svg#icon-aircraft-take-off"></use>
						</svg>
						<span>Flight</span>
					</a>
				</li>
				<li className="side-nav__item">
					<a href="#" className="side-nav__link">
						<svg className="side-nav__icon">
							<use xlinkHref="img/sprite.svg#icon-key"></use>
						</svg>
						<span>Car rental</span>
					</a>
				</li>
				<li className="side-nav__item">
					<a href="#" className="side-nav__link">
						<svg className="side-nav__icon">
							<use xlinkHref="img/sprite.svg#icon-map"></use>
						</svg>
						<span>Tours</span>
					</a>
				</li>
			</ul>

			<div className="legal">
				&copy; 2017 by Trillo. All rights reserved.
			</div>
		</nav>
	)
};

export default Navigation;