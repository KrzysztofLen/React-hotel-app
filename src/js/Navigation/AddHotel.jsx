import React from 'react';
import {Link} from 'react-router-dom';

export const AddHotel = () => (
	<li className="side-nav__item">
		<Link to="/add" className="side-nav__link">
			<span>Add Hotel</span>
		</Link>
	</li>
);