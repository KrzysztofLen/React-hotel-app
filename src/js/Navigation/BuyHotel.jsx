import React from 'react';
import {Link} from 'react-router-dom';

export const BuyHotel = () => (
	<li className="side-nav__item">
		<Link to="/buy" className="side-nav__link">
			<span>Buy Hotel</span>
		</Link>
	</li>
);