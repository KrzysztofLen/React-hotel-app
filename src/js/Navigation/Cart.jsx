import React from 'react';
import {Link} from 'react-router-dom';

export const Cart = () => (
	<li className="side-nav__item">
		<Link to ="/cart" className="side-nav__link">
			<span>Cart</span>
		</Link>
	</li>
);