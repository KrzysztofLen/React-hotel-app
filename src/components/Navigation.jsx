import React, {Component} from 'react';
import {BrowserRouter, NavLink, Link, Route, Router, Switch, Redirect} from 'react-router-dom';
import App from "../App";
import {BuyHotel} from "./BuyHotel/BuyHotel";

class LinkHotel extends React.Component {
	constructor(props) {
		super(props);
		this.onSetActiveIndex = this.onSetActiveIndex.bind(this);
	}

	onSetActiveIndex = () => {
		this.props.onClick(this.props.index);
	};

	componentDidMount() {
		console.log(window.location.href.split("/").pop());
		console.log(this.props.path.split("/").pop());
	}

	// render() {
	// 	return <React.Fragment>
	// 		<li className={this.props.activeIndex ? "side-nav__item side-nav__item--active" : "side-nav__item"} onClick={this.onSetActiveIndex}>
	// 		     {/*onClick={this.onSetActiveIndex}>*/}
	// 			{/*<Link to={`${this.props.href}`} className="side-nav__link">*/}
	// 				{/*<span>{this.props.link}</span>*/}
	// 			{/*</Link>*/}
	// 			<a href="#" className="side-nav__link">
	// 				<span>{this.props.link}</span>
	// 			</a>
	// 		</li>
	// 	</React.Fragment>
	// }

	//TODO Navigation routing
	render() {
		console.log('Active index', this.props.activeIndex);
		return <React.Fragment>
				<li className={this.props.activeIndex ? "side-nav__item side-nav__item--active" : "side-nav__item"}
				    onClick={this.onSetActiveIndex}>
					<NavLink to={`${this.props.path}`} className="side-nav__link">
						<span>{this.props.link}</span>
					</NavLink>
				</li>
			</React.Fragment>
	}
}

class Navigation extends React.Component {
	constructor(props) {
		super(props);
		this.onSetActiveIndex = this.onSetActiveIndex.bind(this);

		this.state = {
			activeIndex: 0
		};
	}

	onSetActiveIndex = (index) => {
		console.log(index);
		this.setState({
			activeIndex: index
		});
	};

	render() {
		// return (
			{/*<nav className="sidebar">*/}
		return (	<ul className="side-nav">
					{routes.map((route, index) => {
						// return <React.Fragment>
						// <LinkHotel key={index} index={index} onClick={this.onSetActiveIndex}
						//                   activeIndex={this.state.activeIndex === index} {...route}/>
						// </React.Fragment>
						return <li className={this.state.activeIndex === index ? "side-nav__item side-nav__item--active" : "side-nav__item"}
						onClick={this.onSetActiveIndex.bind(this, index)}>
					<NavLink to={`${route.path}`} className="side-nav__link">
							<span>{route.link}</span>
					</NavLink>
					</li>
					})}
					{/* TEMPORARY */}
					{/*<li className="side-nav__item side-nav__item--active">*/}
						{/*<a href="/add" className="side-nav__link">*/}
							{/*<span>Add hotel</span>*/}
						{/*</a>*/}
					{/*</li>*/}
					{/* =======*/}
				</ul>)

					{/*return <Route key={index} path={route.path} exact={route.exact} component={route.sidebar}/>*/}
				{/*</Switch>*/}
				{/*<div className="legal">*/}
					{/*&copy; 2017 by Trillo. All rights reserved.*/}
				{/*</div>*/}
			{/*</nav>*/}
		// )
	}
}

Navigation.defaultProps = {
	navlinks: [
		{
			link: "Hotel's",
			path: "/"
		},
		{
			link: "Buy Hotel",
			path: "/buy"
		},
		{
			link: "Cart",
			path: "/cart"
		},
		{
			link: "Add Hotel",
			path: "/add"
		}
	]
};
const routes = [
	{
		link: "Hotel's",
		exact: true,
		path: "/",
		sidebar: App
	},
	{
		link: "Buy Hotel",
		path: "/buy",
		sidebar: BuyHotel
	},
	{
		link: "Cart",
		path: "/cart"
	},
	{
		link: "Add Hotel",
		path: "/add"
	}
];


export default Navigation;