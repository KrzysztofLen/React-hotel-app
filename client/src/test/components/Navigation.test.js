import React from 'react';
import {shallow} from 'enzyme';
import Navigation from "../../components/Navigation/Navigation";

test('should render Navigation component corectlly', () => {
	const wrapper = shallow(<Navigation/>);
	expect(wrapper).toMatchSnapshot();
});