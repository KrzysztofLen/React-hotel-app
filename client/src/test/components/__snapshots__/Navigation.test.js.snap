// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navigation component corectlly 1`] = `
"<Fragment>
  <nav className=\\"side-navigation\\">
    <Logo />
    <ul className=\\"side-navigation__container\\">
      <SideMenuLink activeOnlyWhenExact={true} to=\\"/\\" />
      <SideMenuLink to=\\"/buy\\" />
      <SideMenuLink to=\\"/reservation\\" />
      <SideMenuLink to=\\"/add\\" />
    </ul>
  </nav>
</Fragment>"
`;
